### PR TITLE
OSDOCS-7866: Deploying ROSA in STS mode tutorial.

### PR DESCRIPTION
**Pull request by eohartman**
Version(s):
enterprise-4.13+

Issue:
[OSDOCS-7866](https://issues.redhat.com/browse/OSDOCS-7866)

Link to docs preview:
[Tutorial: Deploying ROSA in STS mode](https://65061--ocpdocs-pr.netlify.app/openshift-rosa/latest/cloud_experts_tutorials/cloud-experts-deploying-rosa-sts-mode)

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+//authors:
+//  - Paul Czarkowski
+//  - Michael Ducy
+// ---
+
+
+* https://youtu.be/R1T0yk9l6Ys[Quick Introduction by Ryan Niksch (AWS) and Shaozen Ding (Red Hat)].
+
+
+STS allows us to deploy ROSA without needing a ROSA admin account, instead it uses roles and policies with Amazon STS (secure token service) to gain access to the AWS resources needed to install and operate the cluster.
```
Comment (mletalie) [inline rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc:26]:
```suggestion
Secure Token Service (STS) allows you to deploy ROSA without needing a ROSA admin account. STS uses roles and policies to gain access to the AWS resources needed to install and operate the cluster.
```

File: rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+. Delete the ROSA cluster.
++
+[source,terminal]
+----
+   $ rosa delete cluster -c $ROSA_CLUSTER_NAME
+----
+. Clean up the STS roles.
++
+Once the cluster has been deleted, you can delete the STS roles. You can get the correct commands with the ID filled in from the output of the previous step.
```
Comment (mletalie) [inline rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc:90]:
```suggestion
Once the cluster has been deleted, you can delete the STS roles. Replace `<id>` with the IDs that were output from the previous step. 
```

**Review by mletalie — COMMENTED** [review]
Done.

File: rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+* xref:../rosa_tutorials/rosa-mobb-prerequisites-tutorial.adoc#rosa-mobb-prerequisites-tutorial[If this is your first time deploying ROSA, follow these prerequisite instructions].
+
+== Deploying a ROSA cluster
+
+. Set environment variables.
++
+[source,terminal]
+----
+   $ export ROSA_CLUSTER_NAME=mycluster
+    export AWS_ACCOUNT_ID=`aws sts get-caller-identity \
```
Comment (yuwang-RH) [inline rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc:38]: the  AWS_ACCOUNT_ID env variable is not used in the following steps.

File: rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      --query Account --output text`
+    export REGION=us-east-2
+    export AWS_PAGER=""
```
Comment (yuwang-RH) [inline rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc:41]: AWS_PAGER is not used in following steps

File: rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+----
+. Make sure your ROSA CLI version is correct (v1.2.25 or above).
++
+[source,terminal]
+----
+   $ rosa version
+----
+. Run the ROSA CLI to create your cluster.
+
```
Comment (yuwang-RH) [inline rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc:50]: Before `rosa create cluster`, account-roles should be created via `rosa create account-roles`

File: rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
++
+[source,terminal]
+----
+   $ rosa create cluster --sts --cluster-name ${ROSA_CLUSTER_NAME} \
+      --region ${REGION} --mode auto --yes
```
Comment (yuwang-RH) [inline rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc:55]: If no flags of account-roles arns are set, the `rosa create cluster ...` command will use the first default bundle of account-roles under the aws account. It's better to set them with specific ones with flags '--role-arn', '--support-role-arn' ,'-worker-iam-role' and '--controlplane-iam-role'

File: rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+----
+. Clean up the STS roles.
++
+Once the cluster has been deleted, you can delete the STS roles. Replace <id> with the IDs from the output of the previous step.
++
+[source,terminal]
+----
+   $ rosa delete operator-roles -c <id> --yes --mode auto
+    rosa delete oidc-provider -c <id>  --yes --mode auto
+----
```
Comment (yuwang-RH) [inline rosa_tutorials/rosa-mobb-deploying-rosa-sts-mode.adoc:96]: There are two flows to create sts cluster, one uses the classic oidc configuration and create the operator-roles for specific cluster, while another flow uses the BYO oidc configuration and the operator-roles prior to cluster spec. 
In this PR, it only introduces the classic flow. Could you confirm if the BYO oidc config+operator-roles prior cluster spec flow needs to be added here. By the way the clean-up steps are using different commands.

**Pull request closed without merging**